tests/unit/require.ts

```diff
@@ -110,6 +120,79 @@ registerSuite({
 		});
 	},
 
+	'non-existent module with on-error listener should not throw error'() {
+		let dfd = this.async(DEFAULT_TIMEOUT);
+		let badMid = 'bad/module/id';
+
+		(<any> process)._events.uncaughtException = function (error: Error) {
+			if (error.message.indexOf(badMid) > -1) {
+				dfd.reject(error);
+			}
+		};
+
+		onErrorHandler = global.require.on('error', dfd.callback(function noop(error: LoaderError) {}));
+
+		global.require([
+			badMid
+		], dfd.rejectOnError(function (gruntOption: any, app: any) {
+			assert.fail(null, null, 'Dependency with bad module id should not be resolved');
```

Comment:
Why not just `throw new Error('Dependency with bad module id should not be resolved');`?